tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -2107,18 +2090,20 @@ uint32_t SPIRVEmitter::getValueZero(QualType type) {
     return theBuilder.getConstantFloat32(0.0);
   }
 
-  if (hlsl::IsHLSLVecType(type)) {
-    const QualType elemType = hlsl::GetHLSLVecElementType(type);
-    const uint32_t elemZeroId = getValueZero(elemType);
+  {
+    QualType elemType = {};
+    uint32_t size = {};
+    if (TypeTranslator::isVectorType(type, &elemType, &size)) {
+      const uint32_t elemZeroId = getValueZero(elemType);
```

Comment:
Actually, we should be able to use `OpConstantNull` for getting the zero value for basically all types (scalar, vector, matrix, struct, etc.). Results in less code and shorter binary.

May change later.